apollo-router/src/query_planner/bridge_query_planner.rs

```diff
@@ -434,45 +427,39 @@ impl Service<QueryPlannerRequest> for BridgeQueryPlanner {
         let fut = async move {
             let start = Instant::now();
 
-            let compiler = match context.private_entries.lock().get::<Compiler>() {
+            let mut doc = match context.private_entries.lock().get::<ParsedDocument>() {
                 None => return Err(QueryPlannerError::SpecError(SpecError::UnknownFileId)),
-                Some(c) => c.0.clone(),
+                Some(d) => d.clone(),
             };
-            let mut compiler_guard = compiler.lock().await;
-            let file_id = compiler_guard
-                .db
-                .source_file(QUERY_EXECUTABLE.into())
-                .ok_or(QueryPlannerError::SpecError(SpecError::UnknownFileId))?;
 
-            match add_defer_labels(file_id, &compiler_guard) {
+            let schema = &this.schema.api_schema().definitions;
+            match add_defer_labels(schema, &doc.ast) {
                 Err(e) => {
                     return Err(QueryPlannerError::SpecError(SpecError::ParsingError(
                         e.to_string(),
                     )))
                 }
                 Ok(modified_query) => {
-                    // We’ve already checked the original query against the configured token limit
-                    // when first parsing it.
-                    // We’ve now serialized a modified query (with labels added) and are about
-                    // to re-parse it, but that’s an internal detail that should not affect
-                    // which original queries are rejected because of the token limit.
-                    compiler_guard.db.set_token_limit(None);
-                    compiler_guard.update_executable(file_id, &modified_query);
+                    doc = Arc::new(ParsedDocumentInner {
+                        executable: modified_query.to_executable(schema),
+                        ast: modified_query,
+                    });
+                    context
+                        .private_entries
+                        .lock()
+                        .insert::<ParsedDocument>(doc.clone());
```

Comment:
why is this added here?

Reply:
This replaces `compiler_guard.update_executable()`. Now that apollo-compiler does not provide interior mutability anymore we rely on the one from `context.private_entries` instead.

Reply:
ok, I see